pkg/util/system/system_utils.go

```diff
@@ -18,14 +18,24 @@ package system
 
 import (
 	"strings"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
 )
 
 // TODO: find a better way of figuring out if given node is a registered master.
-func IsMasterNode(nodeName string) bool {
+// IsMasterNode returns true if given node is a registered master.
+func IsMasterNode(node v1.Node) bool {
```

Comment:
If this is only used in tests it doesn't belong in pkg/util/system.  Move it to test/e2e/...

Reply:
As a general rule, a heuristic like this may not be in the core code base.  It may only be layered on top.

Reply:
Currently, caller just pass a node name and IsMasterNode() check it by string.
IsMasterNode() of my patch need the labels of nodes, so I use v1.node instead of string.

Reply:
@FengyunPan could you move this to test/utils and we can proceed forwards here.

Reply:
It's used in `pkg/controller/node/node_controller.go`